### PR TITLE
`Programming exercises`: Prevent unnecessary page refreshes when opening repositories

### PR DESCRIPTION
**Pull request by pzdr7**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data.

### Motivation and Context

Sometimes, clicking on links or buttons caused a reload of the entire angular app, e.g.

- when opening a LocalVC repository from the participations or the details page of a programming exercise
- when viewing the diff of a commit

This caused longer load times and unnecessary REST calls.

### Description

I changed `[href]` to `[routerLink]` and simplified the links where appropriate, leaving external links as-is (although I did fix some target="_blank"s).

### Steps for Testing
Prerequisites:
- Test server with LocalCI
- 1 Programming exercise
- 1 Instructor

**Open repository (participant)**

1. Log in
2. Navigate to Courses > your course > your programming exercise, click on, and click on "Open Repository".
3. Go to the commit history and click on a commit hash. Verify that the Angular app does not refresh when you do this.

**Open repository (exercise details view)**

5. Navigate to Course management > your course > exercises and click on the title of your programming exercise.
6. For the template, solution, and test repositories: click "Open Repository" and ensure that the Angular app does not refresh when you do so.

**Open repository (participantions)**

7. Still on the details page, open the "Participations" view.
8. Click "Open repository" on any participation. Ensure there is no refresh of the Angular app.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [ ] Test 2

## Summary by CodeRabbit

- **Refactor**
	- Enhanced navigation links in the "Commits Info" component for improved user experience.
	- Streamlined code by removing unused imports and dependencies in various Angular components.
	- Improved routing logic in multiple components for consistent and reliable navigation behavior.

### CODE REVIEW

**Review by krusche — DISMISSED** [review]
code looks good 👍 this is a great improvement, thank you!

**Review by rstief — DISMISSED** [review]
Code LGTM ❤️

**Review by Jan-Thurner — DISMISSED** [review]
Code lgtm

File: src/main/webapp/app/exercises/shared/participation/participation.component.html [hunk 1]
```diff
@@ -143,7 +143,7 @@ <h2>
                             </span>
                         </ng-template>
                         <ng-template ngx-datatable-cell-template let-row="row">
-                            <a [href]="routerLink + '/' + row.id + '/repository'" class="btn btn-sm btn-primary ms-2">
+                            <a [routerLink]="['./', row.id, '/repository']" class="btn btn-sm btn-primary ms-2">
```
Comment (pzdr7) [inline src/main/webapp/app/exercises/shared/participation/participation.component.html:146]:
```suggestion
                            <a [routerLink]="['./', row.id, 'repository']" class="btn btn-sm btn-primary ms-2">
```

**Review by rstief — APPROVED** [review]

**Review by Jan-Thurner — APPROVED** [review]

**Pull request merged**